packages/framework/src/Migrations/Version20191112122617.php

```diff
@@ -0,0 +1,151 @@
+        $this->sql('ALTER TABLE payment_domains ALTER vat_id DROP DEFAULT');
+
+        $this->sql('ALTER TABLE payment_prices DROP CONSTRAINT payment_prices_pkey');
+        $this->sql('ALTER TABLE payment_prices ADD domain_id INT NOT NULL DEFAULT 1');
+        $this->sql('ALTER TABLE payment_prices ALTER domain_id DROP DEFAULT');
+
+        $this->sql('ALTER TABLE transport_domains ADD vat_id INT NOT NULL DEFAULT 1');
+        $this->sql('ALTER TABLE transport_domains ADD CONSTRAINT FK_18AC7F6CB5B63A6B FOREIGN KEY (vat_id) REFERENCES vats (id) NOT DEFERRABLE INITIALLY IMMEDIATE');
+        $this->sql('ALTER TABLE transport_domains ALTER vat_id DROP DEFAULT');
+
+        $this->sql('ALTER TABLE transport_prices DROP CONSTRAINT transport_prices_pkey');
+        $this->sql('ALTER TABLE transport_prices ADD domain_id INT NOT NULL DEFAULT 1');
+        $this->sql('ALTER TABLE transport_prices ALTER domain_id DROP DEFAULT');
+
+        $this->migratePaymentsDomains();
+        $this->migratePaymentsPrices();
+        $this->migrateTransportDomains();
+        $this->migrateTransportPrices();
+
+        $this->sql('ALTER TABLE payment_prices ADD PRIMARY KEY (payment_id, domain_id)');
+        $this->sql('ALTER TABLE transport_prices ADD PRIMARY KEY (transport_id, domain_id)');
+
+        $this->sql('ALTER TABLE payments DROP vat_id');
+        $this->sql('ALTER TABLE transports DROP vat_id');
+
+        $this->sql('ALTER TABLE payment_prices DROP currency_id');
+        $this->sql('ALTER TABLE transport_prices DROP currency_id');
+    }
+
+    /**
+     * @param \Doctrine\DBAL\Schema\Schema $schema
+     */
+    public function down(Schema $schema): void
+    {
+    }
+
+    private function migratePaymentsDomains(): void
+    {
+        $payments = $this->sql('SELECT id, vat_id FROM payments')->fetchAll();
+
+        foreach ($payments as $payment) {
+            $paymentDomains = $this->sql('SELECT id, domain_id FROM payment_domains where payment_id = :paymentId', ['paymentId' => $payment['id']])->fetchAll();
+
+            foreach ($paymentDomains as $paymentDomain) {
+                $vatId = $payment['vat_id'];
+                if ($paymentDomain['domain_id'] > 1) {
+                    $vatId = $this
+                        ->sql('SELECT id FROM vats where tmp_original_id = :tmpOriginalId and domain_id = :domainId', [
+                            'tmpOriginalId' => $payment['vat_id'],
+                            'domainId' => $paymentDomain['domain_id'],
+                        ])
+                        ->fetchColumn(0);
+                }
+
+                $this->sql('UPDATE payment_domain SET vat_id = :vatId WHERE id = :id', [
+                    'vatId' => $vatId,
+                    'id' => $paymentDomain['id'],
+                ]);
+            }
+        }
+    }
+
+    private function migratePaymentsPrices(): void
+    {
+        $paymentPrices = $this->sql('SELECT payment_id, currency_id FROM payment_prices')->fetchAll();
```

Comment:
This will not work when you have two domains with same currency (like german and slovak). Price will be set for only one domain, probably the last one.

Same for transports

Reply:
Pfff, it is hell ...

Reply:
Ok, I changed it